lib/SPIRV/Lowering.md

```diff
@@ -0,0 +1,132 @@
+The SPIRV-LLVM-Translator will "lower" some LLVM intrinsic calls to another function or implementation
```

Comment:
Is it possible to document what logic/criteria is used to select between these three options?

Thanks

Reply:
I've added documentation to explain the pros/cons of each approach.  The approaches are ordered from best to worst in my opinion.